Core/Main/itkElastixRegistrationMethod.hxx

```diff
@@ -979,60 +979,47 @@ template <typename TFixedImage, typename TMovingImage>
 unsigned int
 ElastixRegistrationMethod<TFixedImage, TMovingImage>::GetNumberOfTransforms() const
 {
-  const auto * const transformContainer = m_ElastixMain->GetElastixBase().GetTransformContainer();
-
-  if ((transformContainer == nullptr) || transformContainer->empty())
-  {
-    return 0;
-  }
-
-  const auto * const elxTransformBase =
-    dynamic_cast<ElastixTransformBaseType *>(transformContainer->front().GetPointer());
-
-  return (elxTransformBase == nullptr) ? 0 : elxTransformBase->GetAsITKBaseType()->GetNumberOfTransforms();
+  const auto advancedCombinationTransform = GetAdvancedCombinationTransform();
+  return advancedCombinationTransform ? advancedCombinationTransform->GetNumberOfTransforms() : 0;
 }
 
 
 template <typename TFixedImage, typename TMovingImage>
 auto
 ElastixRegistrationMethod<TFixedImage, TMovingImage>::GetNthTransform(const unsigned int n) const -> TransformType *
 {
-  const auto * const transformContainer = m_ElastixMain->GetElastixBase().GetTransformContainer();
-
-  if ((transformContainer == nullptr) || transformContainer->empty())
-  {
-    return nullptr;
-  }
+  const auto advancedCombinationTransform = GetAdvancedCombinationTransform();
+  return advancedCombinationTransform ? advancedCombinationTransform->GetNthTransform(n) : nullptr;
+}
 
-  const auto * const elxTransformBase =
-    dynamic_cast<ElastixTransformBaseType *>(transformContainer->front().GetPointer());
 
-  if (elxTransformBase == nullptr)
+template <typename TFixedImage, typename TMovingImage>
+auto
+ElastixRegistrationMethod<TFixedImage, TMovingImage>::GetAdvancedCombinationTransform() const
+  -> AdvancedCombinationTransformType *
+{
   {
+    if (m_ElastixMain)
+    {
+      if (const auto transformContainer = m_ElastixMain->GetElastixBase().GetTransformContainer();
+          (transformContainer != nullptr) && !transformContainer->empty())
+      {
+        if (const auto firstTransform = transformContainer->front().GetPointer())
+        {
+          return static_cast<AdvancedCombinationTransformType *>(firstTransform);
```

Comment:
I assume the `static_cast<AdvancedCombinationTransformType *>(firstTransform)` is OK here, casting the first element from `ElastixBase::m_TransformContainer`.